src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
FWIW we have a test that `libcore` compiled in release mode used in debug mode actually results in overflow errors *from within* `libcore` (the `pow` methods IIRC).
Not to mention that the proxy impls for all the operator traits wouldn't have the overflow checks if `libcore` was compiled in release mode.